Terminal.Gui/Core/PosDim.cs

```diff
@@ -167,22 +167,24 @@ public static Pos Center ()
 
 		internal class PosAbsolute : Pos {
 			int n;
+			Func<int> function;
 			public PosAbsolute (int n) { this.n = n; }
+			public PosAbsolute (Func<int> function) { this.function = function; }
```

Comment:
Please could this be a new class e.g. `PosFunc` instead of being part of `PosAbsolute`?  Currently `PosAbsolute` is an absolute unchanging position while others e.g. `PosRelative` are changing.  I think it would be nice to keep these roles if possible.

I know these are private classes but I think we should still follow the single responsibility principle if possible.

Reply:
Thanks for your review. I totally agree with you. I'll create the `PosFunc` class. If the `PosRelative` is more appropriate please let me know.